tests/cache/tests.py

```diff
@@ -2294,22 +2295,49 @@ def test_get_cache_key_with_query(self):
         )
 
 
+def is_fips_mode():
+    '''Test if FIPS mode is enforced'''
+    try:
+        hashlib.md5('fipscheck'.encode())
+    except ValueError as exc:
+        if 'disabled for fips' in exc.message:
+            return 1
+        raise
+    return 0
+
+
 class TestMakeTemplateFragmentKey(SimpleTestCase):
     def test_without_vary_on(self):
         key = make_template_fragment_key('a.fragment')
-        self.assertEqual(key, 'template.cache.a.fragment.d41d8cd98f00b204e9800998ecf8427e')
+        if is_fips_mode():
```

Comment:
As we don't run any Jenkins executors in FIPS mode we could instead use some subtests here and mock the return value of `is_fips_mode` if it's included in `cache/utils.py`?